collect_app/src/test/java/org/odk/collect/android/location/client/MaxAccuracyWithinTimeoutLocationClientTest.java

```diff
@@ -0,0 +1,146 @@
+package org.odk.collect.android.location.client;
+
+import android.location.Location;
+import android.os.Looper;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.odk.collect.android.location.LocationTestUtils.createLocation;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public class MaxAccuracyWithinTimeoutLocationClientTest {
+    private FakeLocationClient fakeLocationClient;
+    private TestLocationListener testLocationListener;
+
+    private MaxAccuracyWithinTimeoutLocationClient maxAccuracyLocationClient;
+
+    @Before
+    public void setUp() {
+        fakeLocationClient = new FakeLocationClient();
+        testLocationListener = new TestLocationListener();
+
+        maxAccuracyLocationClient = new MaxAccuracyWithinTimeoutLocationClient(fakeLocationClient, testLocationListener);
+    }
+
+    @Test
+    public void requestingLocationUpdates_ShouldResultInUpdate() {
```

Comment:
Tiny nitpick: Especially given the length of test names in JUnit I think we can leave out "should" (so just `requestingLocationUpdates_resultsInUpdate`)?

Reply:
Aw, I like my shoulds! But I see your point. c70eb98